src/cursor/core_cursor.ts

```diff
@@ -176,93 +173,65 @@ export class CoreCursor<
     }
 
     // All internal state
-    this.cursorState = {
-      cursorId,
-      cmd: this.cmd,
-      lastCursorId,
-      documents: options.documents || [],
-      cursorIndex: 0,
-      dead: false,
-      killed: false,
-      init: false,
-      notified: false,
-      limit,
-      skip,
-      batchSize,
-      currentLimit: 0,
-      // Result field name if not a cursor (contains the array of results)
-      transforms: options.transforms,
-      raw: typeof options.raw === 'boolean' ? options.raw : cmd && 'raw' in cmd && cmd.raw,
-      tailable: typeof options.tailable === 'boolean' ? options.tailable : false,
-      awaitData:
-        typeof options.awaitData === 'boolean'
-          ? options.awaitData
-          : typeof options.awaitdata === 'boolean'
-          ? options.awaitdata
-          : false
-    };
+    this.cursorId = cursorId;
+    this.lastCursorId = lastCursorId;
+    this.documents = options.documents || [];
+    this.cursorIndex = 0;
+    this.dead = false;
+    this.killed = false;
+    this.init = false;
+    this.notified = false;
+    this.currentLimit = 0;
+    // Result field name if not a cursor (contains the array of results)
+    this.transforms = options.transforms;
+    this.raw = typeof options.raw === 'boolean' ? options.raw : cmd && 'raw' in cmd && cmd.raw;
+    this.tailable = typeof options.tailable === 'boolean' ? options.tailable : false;
+    this.awaitData =
+      typeof options.awaitData === 'boolean'
+        ? options.awaitData
+        : typeof options.awaitdata === 'boolean'
+        ? options.awaitdata
+        : false;
+
+    // get rid of these?
+    this._limit = limit;
```

Comment:
`_limit` and `_batchSize` seem to be used. What's the purpose of this comment and the deprecation  comment above? Are these meant to be `private`? If so can we use private keyword?

Reply:
They are used, but won't be for much longe (referring to the cursor prototype I discussed in our meeting yesterday). For now, I wanted to keep these around to make @emadum 's work of combining `Cursor` and `CoreCursor` easier.